src/FileType_HDF5.f90

```diff
@@ -6571,5 +6595,338 @@ SUBROUTINE compute_chunk_size(gdims,cdims)
+!> @param attr_value the desired value of the attrbute
+!>  
+#ifdef FUTILITY_HAVE_HDF5
+    SUBROUTINE read_attribute_st0(this,attr_id,length_max,attr_name,attr_val)
+       CHARACTER(LEN=*),PARAMETER :: myName='read_attribute_st0_HDF5FileType'
+       CLASS(HDF5FileType),INTENT(INOUT) :: this
+       CHARACTER(LEN=*),INTENT(IN) :: attr_name
+       TYPE(StringType),INTENT(INOUT)::attr_val
+       INTEGER(SDK),INTENT(IN) :: length_max
+        
+       INTEGER(HID_T),INTENT(IN) :: attr_id
+       INTEGER(HID_T)::atype_id
+       INTEGER(HSIZE_T),DIMENSION(1) :: dims
+       CHARACTER(LEN=length_max,KIND=C_CHAR),TARGET :: buf
+
+       dims(1)=1
+       CALL h5aget_type_f(attr_id,atype_id,error)
+       CALL h5aread_f(attr_id,atype_id,buf,dims,error)
+       attr_val=buf
+    END SUBROUTINE read_attribute_st0
+#endif
+!
+!-------------------------------------------------------------------------------
+!> @brief Reads a integer value attribute from a known dataset
+!> 
+!> @param obj_name the relative path to the dataset
+!> @param attr_name the desired name of the attribute
+!> @param attr_value the desired value of the attrbute
+!>  
+    SUBROUTINE read_attribure_i0(this,obj_name,attr_name,attr_val)
+       CHARACTER(LEN=*),PARAMETER :: myName='read_attribute_i0_HDF5FileType'
+       CLASS(HDF5FileType),INTENT(INOUT) :: this
+       CHARACTER(LEN=*),INTENT(IN) :: obj_name, attr_name
+       INTEGER(SNK),INTENT(INOUT) :: attr_val
+        
+#ifdef FUTILITY_HAVE_HDF5
+       INTEGER(HID_T) :: attr_id, dset_id
+       INTEGER(HSIZE_T),DIMENSION(1) :: dims
+       dims(1)=1
+
+       !Prepare the File and object for the attribute
+       CALL open_object(this,obj_name,dset_id)
+       CALL pre_read_attribute(this,dset_id,attr_name,attr_id)
+
+       CALL h5aread_f(attr_id,H5T_NATIVE_INTEGER,attr_val,dims,error)
+       IF(error /= 0) THEN 
+         CALL this%e%raiseError(modName//'::'//myName// &
+          ' - Failed to read attribute.')
+         RETURN
+       ENDIF
+
+       CALL post_attribute(this,attr_id,dset_id)
+#endif
+    END SUBROUTINE read_attribure_i0
+!
+!-------------------------------------------------------------------------------
+!> @brief Reads a double value attribute from a known dataset
+!> 
+!> @param obj_name the relative path to the dataset
+!> @param attr_name the desired name of the attribute
+!> @param attr_value the desired value of the attrbute
+!>  
+    SUBROUTINE read_attribute_d0(this,obj_name,attr_name,attr_val)
+       CHARACTER(LEN=*),PARAMETER :: myName='read_attribute_d0_HDF5FileType'
+       CLASS(HDF5FileType),INTENT(INOUT) :: this
+       CHARACTER(LEN=*),INTENT(IN) :: obj_name, attr_name
+       REAL(SDK),INTENT(INOUT) :: attr_val
+        
+#ifdef FUTILITY_HAVE_HDF5
+       INTEGER(HID_T) :: attr_id, dset_id
+       INTEGER(HSIZE_T),DIMENSION(1) :: dims
+       dims(1)=1
+
+       !Prepare the File and object for the attribute
+       CALL open_object(this,obj_name,dset_id)
```

Comment:
This is pretty nit-picky, but if you are going to open_object explicitly outside of the pre_read_attribute routine, it looks scary not to see a matching close_object call. I would recommend calling open_object inside the pre_read_attribute, so that it's more clear that things are getting cleaned up right.